src/components/Comments.jsx

```diff
@@ -0,0 +1,54 @@
+import { useState,useEffect } from "react";
+import { useParams } from "react-router-dom";
+import { getCommentsByArticleId } from "../api";
+
+function Comments() {
+    const { article_id } = useParams();
+    const [commentsByArticle, setCommentsByArticle] = useState([])
+    const [isLoading, setIsLoading] = useState(false)
+    const [isError, setIsError] = useState(false)
+    useEffect(() => {
+        setIsLoading(true)
+        getCommentsByArticleId(article_id).then((comments) => {
+            setCommentsByArticle(comments)
+            setIsLoading(false)
+            setIsError(false)
+        }).catch((err) => {
+            setIsLoading(false)
+            setIsError(true)
+          });
+    }, [article_id])
+
+    if(isLoading){
+        return (
+            <p>Loading Comments...</p>
+        )
+    }
+    if(isError){
+        return (
+            <p>Loading Comments...</p>
+        )
```

Comment:
Something you could consider is extracting your Loading state out and making it a reusable component